### ISSUE TITLE
Closure under composition

### ISSUE THREAD

**Issue by pietercollins** [ariadne-cps/ariadne#378]
Although the class of hybrid automata is closed under parallel composition, there is no way of obtaining a `HybridAutomaton` as the result of a composition.

We should provide a _flatten_  operation to obtain a  `HybridAutomaton` from the `CompositeHybridAutomaton` constructed from the composition. We should also provide a parallel_composition for `CompositeHybridAutomaton` objects.

**Comment by lgeretti** [comment]
I agree on the latter (should we provide a corresponding constructor too?).

Regarding the former: is it worth it? Do we get any practical advantage from flattening?
